### PR TITLE
[Bug] [lang] Fix copying Matrix/StructField elements in Taichi scope

### PR DESCRIPTION
**Pull request by strongoier**
Related issue = #3625

Accidentally, assignments from Matrix/StructField elements to local variables are by reference:
```python
import taichi as ti

ti.init(debug=True)

a = ti.Vector.field(3, ti.i32, shape=())
a[None] = ti.Vector([1, 2, 3])

@ti.kernel
def test():
    b = a[None]
    b[0] = 5
    b[1] = 9
    b[2] = 7
    print(a[None][0])  # 5
    print(a[None][1])  # 9
    print(a[None][2])  # 7

test()
```

However, they are indented to be by value. The PR aims at fixing it. After this PR, https://github.com/taichiCourse01/taichi_ray_tracing/blob/master/4_0_path_tracing.py should work fine.

### CODE REVIEW

**Review by lin-hitonami — APPROVED** [review]
LGTM!

File: tests/python/test_custom_struct.py [hunk 1]
```diff
@@ -315,3 +315,24 @@ def test():
         assert b.b == 4
 
     test()
+
+
+@ti.test(debug=True)
+def test_copy_struct_field_element_to_taichi_scope():
```
Comment (ailzhang) [inline tests/python/test_custom_struct.py:321]: OOC: why do we need to set `debug=True` here? ;)

**Review by ailzhang — APPROVED** [review]
LGTM!

File: tests/python/test_custom_struct.py (hunk 1 above)
Comment (strongoier) [reply tests/python/test_custom_struct.py:321]: Because assertions in kernels are enabled only when `debug=True` :-)

**Pull request merged**